test/e2e/network_policy.go

```diff
@@ -0,0 +1,438 @@
+		if !framework.ProviderIs("local") {
+			framework.Failf(fmt.Sprintf("The test requires two Ready nodes on %v, but found just one.", framework.TestContext.Provider))
+		}
+		framework.Logf("Only one ready node is detected. The test has limited scope in such setting. " +
+			"Rerun it with at least two nodes to get complete coverage.")
+	}
+
+	// Create service A and B in namespaces A and B respectively.  The services are used
+	// for pod discovery by the net monitor containers.
+	serviceA := createNetworkPolicyService(f, nsA, serviceAName)
+	serviceB := createNetworkPolicyService(f, nsB, serviceBName)
+
+	// Clean up services
+	defer func() {
+		By("Cleaning up the service A")
+		if err = f.Client.Services(nsA.Name).Delete(serviceA.Name); err != nil {
+			framework.Failf("unable to delete svc %v: %v", serviceA.Name, err)
+		}
+	}()
+	defer func() {
+		By("Cleaning up the service B")
+		if err = f.Client.Services(nsB.Name).Delete(serviceB.Name); err != nil {
+			framework.Failf("unable to delete svc %v: %v", serviceB.Name, err)
+		}
+	}()
+
+	By("Creating a webserver (pending) pod on each node")
+	podA, podBs := launchNetMonitorPods(f, nsA, nsB, nodes)
+
+	// Deferred clean up of the pods.
+	defer func() {
+		By("Cleaning up the webserver pods")
+		if err = f.Client.Pods(nsA.Name).Delete(podA.Name, nil); err != nil {
+			framework.Logf("Failed to delete pod %v: %v", podA.Name, err)
+		}
+		for _, podB := range podBs {
+			if err = f.Client.Pods(nsB.Name).Delete(podB.Name, nil); err != nil {
+				framework.Logf("Failed to delete pod %v: %v", podB.Name, err)
+			}
+		}
+	}()
+
+	// Wait for all pods to be running.
+	By(fmt.Sprintf("Waiting for pod %v to be running", podA.Name))
+	err = framework.WaitForPodRunningInNamespace(f.Client, podA)
+	Expect(err).NotTo(HaveOccurred())
+	for _, podB := range podBs {
+		By(fmt.Sprintf("Waiting for pod %v to be running", podB.Name))
+		err = framework.WaitForPodRunningInNamespace(f.Client, podB)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// Open up TCP port used to access the network monitor HTTP API (8080).  Until we do this, all
+	// pods in all namespaces are isolated and the test cannot query the netmonitor pods to pull
+	// the connectivity information.
+	By("Open up monitoring port so that test can pull connectivity data from services")
+	addNetworkPolicyOpenPort(f, nsA, "query1", 8080, api.ProtocolTCP)
+	addNetworkPolicyOpenPort(f, nsB, "query2", 8080, api.ProtocolTCP)
+
+	// We expect full isolation between all pods in all namespaces.  Monitor, wait for a little and
+	// re-check we are still isolated.
+	By("Checking full isolation")
+	expected := Summary{
+		TCPNumOutboundConnected: 0,
+		TCPNumInboundConnected:  0,
+	}
+	monitorConnectivity(f, nsA, serviceA.Name, expected)
+	monitorConnectivity(f, nsB, serviceB.Name, expected)
+
+	By("Waiting and rechecking full isolation")
+	time.Sleep(10 * time.Second)
+	monitorConnectivity(f, nsA, serviceA.Name, expected)
```

Comment:
I think because this was the first connectivity check performed I wanted to make sure it wasn't just a transient condition, but I'm not sure it's really buying much other than slowing the test down - I'll remove.